packages/charts/react-charting/UnitTests/PlotlySchemaAdapterUT.test.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  isDateArray,
+  isNumberArray,
+  isMonthArray,
+  updateXValues,
+  getColor,
+  transformPlotlyJsonToDonutProps,
+  transformPlotlyJsonToVSBCProps,
+  transformPlotlyJsonToGVBCProps,
+  transformPlotlyJsonToVBCProps,
+  transformPlotlyJsonToScatterChartProps,
+  transformPlotlyJsonToHorizontalBarWithAxisProps,
+  transformPlotlyJsonToHeatmapProps,
+  transformPlotlyJsonToSankeyProps,
+  transformPlotlyJsonToGaugeProps,
+  sanitizeJson,
+} from '../src/components/DeclarativeChart/PlotlySchemaAdapter';
+
+const colorMap = new Map<string, string>();
+const env = require('../config/tests');
+const runTest = env === 'TEST' ? describe : describe.skip;
+
+runTest('isDate', () => {
+  test('Should return true when input array contains Date objects', () => {
+    var date = new Date();
+    expect(isDateArray([date, date.getDate() + 1, date.getDate() + 2])).toBe(true);
+  });
+
+  test('Should return false when input array contains numeric data', () => {
+    expect(isDateArray([20, 30, 40])).toBe(false);
+  });
+
+  test('Should return false when input array contains string data', () => {
+    expect(isDateArray(['twenty', 'thirty', 'forty'])).toBe(false);
+  });
+});
+
+runTest('isNumberArray', () => {
+  test('Should return false when input array contains Date objects', () => {
+    var date = new Date();
+    expect(isNumberArray([date, date.getDate() + 1, date.getDate() + 2])).toBe(false);
+  });
+
+  test('Should return true when input array contains numeric data', () => {
+    expect(isNumberArray([20, 30, 40])).toBe(true);
+  });
+
+  test('Should return true when input array contains numaric data in string formatt', () => {
+    expect(isNumberArray(['20', '30', '40'])).toBe(true);
+  });
+
+  test('Should return false when input array contains string data', () => {
+    expect(isNumberArray(['twenty', 'thirty', 'forty'])).toBe(false);
+  });
+});
+
+runTest('isMonthArray', () => {
+  test('Should return false when input array contains Date objects', () => {
+    var date = new Date();
+    expect(isMonthArray([date, date.getDate() + 1, date.getDate() + 2])).toBe(false);
+  });
+
+  test('Should return true when input array contains months data', () => {
+    expect(isMonthArray([10, 11, 1])).toBe(true);
```

Comment:
pass actual month values - "January", "Jan" etc.
Also add test case for months in other locale like spanish, italian, chinese